api/v1alpha1/bundle_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type BundleSourceType string
+
+// +union
+type BundleSource struct {
+	// +unionDiscriminator
+	// +optional
+	Type BundleSourceType
```

Comment:
What does the `type` tell us that we can't derive from the `class`? Do you have examples in mind?

Reply:
The idea was to make room for sources of bundle content that we couldn't express as a URI-like reference; e.g. embedded yaml or maybe volume mounts.